lib/improver/weighted_blend.py

```diff
@@ -33,13 +33,208 @@
+                                perc_dim=perc_dim)
+        return result
+
+    @staticmethod
+    def blend_percentile_aggregate(data, axis,
+                                   arr_percent, arr_weights, perc_dim):
+        """ Blend percentile aggregate function
+
+        Args:
+            data : np.array
+                   Array containing the data to blend
+            axis : integer
+                   The index of the coordinate dimension in the cube.
+            arr_percent: np.array
+                     Array of percentile values e.g
+                     [0, 20.0, 50.0, 70.0, 100.0],
+                     same size as the percentile dimension of data.
+            arr_weights: np.array
+                     Array of weights, same size as the axis dimension of data.
+            perc_dim : integer
+                     The index of the perecentile coordinate
+            (Note percent and weights have special meaning in Aggregator
+             hence the rename.)
+
+        Returns:
+            result : np.array
+                     containing the weighted percentile blend data
+                     across the chosen coord
+        """
+        if axis < 0:
+            axis += data.ndim
+        # Firstly ensure axis coordinate and percentile coordinate
+        # are indexed as the first and second values in the data array
+
+        data = np.moveaxis(data, [perc_dim, axis], [1, 0])
+
+        # Determine the rest of the shape
+        shape = data.shape[2:]
+        result = None
+        if shape:
+            input_shape = [data.shape[0],
+                           data.shape[1],
+                           np.prod(shape)]
+            # Flatten the data that is not percentile or coord data
+            data = data.reshape(input_shape)
+            # Create the resulting data array
+            result = np.zeros(input_shape[1:])
+            # Loop over the flatten data
+            for i in range(data.shape[-1]):
+                result[:, i] = (
+                    BlendingUtilities.blend_percentiles(data[:, :, i],
+                                                        arr_percent,
+                                                        arr_weights))
+        # Reshape the data and put the percentile dimension
+        # back in the right place
+        if arr_percent.shape > (1,):
+            shape = arr_percent.shape + shape
+            result = result.reshape(shape)
+            if axis < perc_dim:
+                if perc_dim != 1:
+                    result = np.moveaxis(result, 0, perc_dim-1)
+            else:
+                if perc_dim != 0:
+                    result = np.moveaxis(result, 0, perc_dim)
+        return result
+
+    @staticmethod
+    def blend_percentiles(perc_values, percentiles, weights):
+        """ Blend percentiles function
+
+        Args:
+            perc_values : np.array
+                   Array containing the percentile values to blend
```

Comment:
Changed to this: 

`Array containing the percentile values to blend, with
shape (length of coord to blend, num of percentiles)`